presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveWriteUtils.java

```diff
@@ -524,7 +525,7 @@ public static Path createTemporaryPath(ConnectorSession session, HdfsContext con
 
         // use relative temporary directory on ViewFS
         if (isViewFileSystem(context, hdfsEnvironment, targetPath)) {
-            temporaryPrefix = ".hive-staging";
+            temporaryPrefix = new HiveConfig().getViewFsTemporaryStagingDirectoryPath();
```

Comment:
In order for the config to propagate correctly, we'll need to add `HiveConfig` during `LocationService` construction and provide to this method.